dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/DataProvider.java

```diff
@@ -403,13 +408,53 @@ public void componentMetaDataCreated(MetaDataEvent<ComponentMetaData> event) {
         ComponentMetaData metadata = event.getMetaData();
         J2EEName jeeName = metadata.getJ2EEName();
 
+        boolean isEJBmetadata = metadata instanceof IdentifiableComponentMetaData ? ((IdentifiableComponentMetaData) metadata).getPersistentIdentifier().startsWith("EJB") : false;
```

Comment:
If you want a more concise way of writing it,

```suggestion
        boolean isEJBmetadata = metadata instanceof IdentifiableComponentMetaData i &&
                                i.getPersistentIdentifier().startsWith("EJB");
```